crates/matrix-sdk-ffi/src/lib.rs

```diff
@@ -82,6 +86,28 @@ fn new_client_builder(base_path: String, home: String) -> anyhow::Result<ClientB
     Ok(MatrixClient::builder().user_agent("rust-sdk-ios").store_config(store_config))
 }
 
+fn new_client_builder_with_config(base_path: String, home: String, config: ClientConfig, user: &UserId) -> anyhow::Result<ClientBuilder> {
+    let builder = new_client_builder(base_path, home)?;
+
+    let builder = match config.homeserver {
```

Comment:
Is there a rust-ier way to write this, i.e. using builder methods based on optional parameters and thus repeatedly shadowing the `builder` local variable?

Reply:
I'd do this via mutation:

```rust
let mut builder = new_client_builder(base_path, home)?;

if let Some(homeserver) = config.homeserver {
    builder = builder.homeserver_url(&homeserver);
} else {
    builder = builder.user_id_server(&user);
}
```